lib/server/mapper.js

```diff
@@ -126,6 +129,24 @@ class DataMapper extends EventEmitter {
               let d = {};
               d.file        = f.filepath_by_host[host]  || f.filepath_by_host["*"];
               d.accessGroup = f.access_control_group_id || '';
+
+              if (f.avh.reference) {
+                // Reference path in database is often incorrect. So use only
+                // sample name and filename from database's path, and instead
+                // substitute in actual path.
+                let ref   = f.avh.reference.split('/').slice(-5);
+                ref[ref.length - 2] = 'fasta';
+                ref                 = ref.join('/');
+                let options = config.provide();
+                let rootref = options.get('references');
+                console.log(rootref);
+                if (rootref) {
```

Comment:
This bit is the tricky one because I don't think it will work in tests because it tries to check file system, but we can try to go around it.